### PR TITLE
Added ImpactRotation and ImpactPosition to WarheadArgs

### PR DESCRIPTION
**Pull request by reaperrr**
See commits for more details.

Mostly for downstream projects for now.

I'll provide an example testcase soon (probably tomorrow).

### CODE REVIEW

**Comment by Mailaender** [comment]
Both `ImpactPosition` and `ImpactRotation` seem to be unused. Why do you want to set those at time of impact? Wouldn't it make more sense for a projectile to get rotation at launch to match the launcher or rotate it while it is in the air and flying an arch?

**Comment by reaperrr** [comment]
Because for warheads, only the projectile position and rotation at the moment of impact matters, I don't see the point in setting them earlier or updating on the fly (that would just cost performance).

I'm using them downstream for my directional armor implementation, which I might submit upstream after this (not sure if my approach would get conceptual approval, but maybe it will).

**Comment by reaperrr** [comment]
Note: I'll hold off rebasing this until #18023 is merged (which hopefully won't take too much longer).

File: OpenRA.Game/GameRules/WeaponInfo.cs
Comment (pchote) [inline OpenRA.Game/GameRules/WeaponInfo.cs]: Would `ImpactOrientation` still make sense here? This would make it consistent with the actor orientation naming.

**Review by pchote — COMMENTED** [review]
LGTM, just one naming comment/question.

I haven't tried testing this yet because this doesn't touch anything used in the upstream mods.

**Comment by reaperrr** [comment]
Rebased and updated.

Note: Due to rebase conflicts caused by the recent Facing int->WAngle changes, I decided to merge the first 2 commits to save me from more rebase conflict hell.

**Review by teinarss — APPROVED** [review]

**Comment by teinarss** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/c04fd6c08509a8386fcf23380740bdabb1a5f6eb...7e2eaa697bd84c1bb670606e1fcb0ac030923f5d

**Pull request merged**
